Packs/AzureFirewall/Integrations/AzureFirewall/AzureFirewall.py

```diff
@@ -0,0 +1,2848 @@
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+
+    """
+    if output_key:
+        outputs = copy.deepcopy(response.get(output_key, []))
+    else:
+        outputs = copy.deepcopy(response)
+
+    if not isinstance(outputs, list):
+        outputs = [outputs]
+
+    readable_data = []
+
+    for firewall in outputs:
+        properties = firewall.get("properties", {})
+        ip_configuration = properties.get("ipConfigurations", [])
+        ip_configuration = ip_configuration[0] if ip_configuration else {}
+        data = {
+            "name": firewall.get("name"),
+            "id": firewall.get("id"),
+            "location": firewall.get("location"),
+            "threat_intel_mode": properties.get("threatIntelMode"),
+            "private_ip_address": dict_safe_get(ip_configuration, ["properties", "privateIPAddress"]),
+            "subnet": dict_safe_get(ip_configuration, ["properties", "subnet", "id"]),
+            "provisioning_state": properties.get("provisioningState")
+        }
+
+        readable_data.append(data)
+
+    readable_output = tableToMarkdown(
+        readable_header,
+        readable_data,
+        headers=['name', 'id', 'location', 'subnet', 'threat_intel_mode', 'private_ip_address', 'provisioning_state'],
+        headerTransform=string_to_table_header
+    )
+
+    command_results = CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='AzureFirewall.Firewall',
+        outputs_key_field='id',
+        outputs=outputs,
+        raw_response=response
+    )
+
+    return command_results
+
+
+def azure_firewall_list_command(client: AzureFirewallClient, args: Dict[str, Any]) -> CommandResults:
+    """
+    List azure firewalls in resource group or subscription.
+    Args:
+        client (AzureFirewallClient): Azure Firewall API client.
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+
+    """
+    resource = args.get('resource', 'resource_group')
+    limit = arg_to_number(args.get('limit') or '50')
+    page = arg_to_number(args.get('page') or '1')
+    validate_pagination_arguments(limit, page)
+
+    readable_message = get_pagination_readable_message(header='Firewall List:',
+                                                       limit=limit, page=page)
+
+    start_offset = (page - 1) * limit
+    end_offset = start_offset + limit
+    complete_requests = False
+    total_response = {'value': []}
+    response = client.azure_firewall_list_request(resource=resource)
+
+    while not complete_requests:
+        total_response['value'].extend(response.get('value'))
+        if len(total_response['value']) >= end_offset or not response.get('nextLink'):
+            complete_requests = True
+        else:
+            response = client.azure_firewall_list_request(resource=resource, next_link=response.get('nextLink'))
```

Comment:
if the API has paging handling(next link) shouldn't we be able to pass the page six and number, or limit/skip to the api? this way we will not need fill out the pages, just get to the next link.

Reply:
As we agreed with @meital-aizen  in our previous integrations - The pagination arguments should be `page` and `limit` - and we have to handle the code according to the API limitations and behavior.